### PR TITLE
chore(internal): use `@ts-expect-error` directive instead

### PR DESCRIPTION
**Pull request by seia-soto**
- Removes `@typescript-eslint/ban-ts-comment`.

Notes

We need to figure out why `fastHash` is handling an empty array: `[]`.

### CODE REVIEW

**Review by chrmod — COMMENTED** [review]
Not super sure if this is any improvement

File: packages/adblocker-extended-selectors/test/unit/eval.test.ts [hunk 1]
```diff
@@ -27,13 +27,9 @@ function testMatches(selector: string, html: string, target: string, expected: b
...
   }
 
-  const element = target === 'document' ? document : document.querySelector(target);
-  expect(element).to.not.be.null;
-  if (element !== null) {
-    // NOTE: here we need to ignore the type warnings so that we can pass a
-    // `Document` argument to test some edge cases (e.g. textContent returns
-    // null on document).
-    // @ts-ignore
+  const element = document.querySelector(target);
```
Comment (chrmod) [inline packages/adblocker-extended-selectors/test/unit/eval.test.ts]: Removing the `@ts-ignore` exposed a wrong assumption. `matches` is incapable of accepting `document` as it is not an Element. If needed it should be able to handle a `documentElement` which is a valid Element. Thanks to this realization we can remove two invalid tests.

Comment (seia-soto) [reply packages/adblocker-extended-selectors/test/unit/eval.test.ts]: According to [`0c3f9e6` (#4974)](https://github.com/ghostery/adblocker/pull/4974/commits/0c3f9e6aa6c4bb2696b18a0d9c2d883a5e0b8e82) and [`058b36b` (#4974)](https://github.com/ghostery/adblocker/pull/4974/commits/058b36baf647818b476387cd27a168b3c7e5fafc), I also wondered about the exact spec. However, we need to understand that if we head to that direction, `html` (pure CSS selector) will not work anymore and this may be misleading. Still pseudo-class selectors like `:upward(html)` will continue to work since they're traversing via `HTMLElement.prototype.parentElement`.

`Document` is not compatible with `HTMLHtmlElement` but please ensure that there's no side-effect. By looking at the test, I believe @remusao tried to support it as well by following the behaviors of `document.querySelector` despite our own `querySelectorAll` is accepting `Element` only (reasonably `HTMLElement`).

File: packages/adblocker/src/data-view.ts [hunk 2]
```diff
@@ -429,8 +429,10 @@ export class StaticDataView {
     const byteLength = this.getLength();
     this.pos += byteLength;
 
-    // @ts-ignore
-    return String.fromCharCode.apply(null, this.buffer.subarray(this.pos - byteLength, this.pos));
+    return String.fromCharCode.apply(
+      null,
+      Array.from(this.buffer.subarray(this.pos - byteLength, this.pos)),
+    );
```
Comment (seia-soto) [inline packages/adblocker/src/data-view.ts]: I don't think this changes are necessary and actually helping. It's pretty clear that `Array.from` adds an additional loop on top. I'll put benchmark results here later.

**Comment by seia-soto** [comment]
I removed all source code changes and kept existing behaviors, creating a new issue is recommended.
